lib/abide_dev_utils/xccdf.rb

```diff
@@ -298,7 +298,9 @@ def map_indexed(index: 'title', framework: 'cis', key_prefix: '')
 
       def facter_platform
         cpe = xpath('xccdf:Benchmark/xccdf:platform')[0]['idref'].split(':')
-        [cpe[4].split('_')[0], cpe[5].split('.')[0]]
+        return [cpe[4].split('_')[0], cpe[5].split('.')[0]] if cpe.length > 4
+
+        [cpe[3].split('_')[0], '']
```

Comment:
This won't work. This function is supposed to return an array with two items: the OS name, and the OS major version. Returning a major version of `''` can have unintended side effects and should be avoided.

Reply:
Would returning a `-` to stub that array works? Since this function is for creating the directory for the mapped out YAML.

Reply:
No, we need to get the `10` from `Windows_10` as the second item in that array. If we don't it will throw off the Hiera lookup path.

Reply:
So for Windows Server 2019, this is the cpe: `cpe:2.3:o:microsoft:windows_server_2019:-:*:*:*:*:*:*:*`.  Wouldn't the current main branch code return ['microsoft', '-'] from this function? That would also throw off the Hiera lookup path.

Reply:
Yep, it would. Here's how that function gets used: https://github.com/puppetlabs/abide_dev_utils/blob/e6f19183770052c01a1eba94b889f9e1470ee111/lib/abide_dev_utils/xccdf.rb#L235. Looks like the actual root cause of the bug. I wonder if CIS changed up their CPEs in the latest benchmarks? Seems strange that this wouldn't have bit us before.

Reply:
I can add more logic to check the 5th index if it's not a `-`.  Since for RedHat xml, the 5th index of the array is the version number.

Reply:
That should work. I'd also check to see if the 5th index has any value at all. We should also make a note of this and open a ticket with CIS because these CPEs are malformed.

Reply:
Ok. I'll make it a note in the ticket.